src/core/utils/urlutils.js

```diff
@@ -43,19 +43,26 @@ export function getAnalyticsUrl (env = PRODUCTION, conversionTrackingEnabled = f
 
 /**
  * Returns the passed in url, with all url params from the current url, as well as any
- * pasased in params, appended to it.
- * @param {string} url
+ * passed in params, appended to it.
+ * @param {string} url Full url, if there are query params in the url, they are removed
  * @param {Object} params
+ * @param {SearchParams} currentParams The params of the current URL, kept in the final url
  * @returns {string}
  */
-export function addParamsToUrl (url, params = {}) {
-  const urlParams = new SearchParams(window.location.search.substring(1));
+export function addParamsToUrl (url, params = {}, currentParams) {
```

Comment:
This new function signature is a bit confusing. It feels like params and currentParams are doing pretty similar things. Both are for specifying params that will end up in the final URL. Their differentiator is the type? currentParams is typeof SearchParams and params is an object. 

Instead of this, could we mock new SearchParams in Jest? I think that would make the unit tests a bit more accurate as well, since we'd be ensuring that calls to this method invoke new SearchParams properly. All real calls to this method will be using new SearchParams and not passing in currentParams.

https://stackoverflow.com/questions/52886079/mocking-urlseachparams

Reply:
Ah, I agree, the function signature is confusing.

Originally, this was added so that this function is not always relying on a global variable (ie `window.location.search`). This is so we do not have to do [this hack](https://github.com/yext/answers/blob/master/tests/core/storage/persistentstorage.js#L9) to get that variable modifiable by the code.

I think ideally, this util function would not rely on the current URL params and instead just apply the params to the given URL. I did not want to refactor everything, in case things were relying on this fact, though. I think we set ourselves up for unknown side-effects if we are relying on this, though, because the URL changes with any call to persistentStorage. 

What do you think? Should we refactor the function and make it so that params is `SearchParams` instead and includes all of the current params (so that way it is easier to reproduce behavior and makes unit tests easier)? Or should that be another item and instead use hack above?

Reply:
I think the refactor you're proposing would be good! This method takes in a baseUrl and adds whatever is in params to it. If you want all the current URL params added, ensure they are part of params. There are no surprises that way, we add exactly what you ask us to.

 If it's easier for now, I'd say lets keep the method as it was (with the previous signature) and add the hack(s) to the Jest tests. We can make an item for the refactor.

Reply:
This has been refactored to help with another item!